src/rules/orderedImportsRule.ts

```diff
@@ -0,0 +1,125 @@
+import * as Lint from "../lint";
+import * as ts from "typescript";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "ordered-imports",
+        description: "Requires that import statements be alphabetized.",
+        descriptionDetails: Lint.Utils.dedent`
+            Enforce a consistent ordering for ES6 imports:
+            - Named imports must be alphabetized (i.e. "import {A, B, C} from "foo";")
+                - The ordering is alphabetical, with lowercase before uppercase.
+            - Import sources must be alphabetized within groups, i.e.:
+                    import * as a from "a";
+                    import * as a from "b";
+            - groups of imports are delineated by blank lines. You can use these to group imports
+                however you like, e.g. by first- vs. third-party or thematically.`,
+        optionsDescription: "Not configurable.",
+        options: null,
+        optionExamples: ["true"],
+        type: "style",
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static NAMED_IMPORTS_UNORDERED = "Named imports must be alphabetized.";
+    public static IMPORT_SOURCES_UNORDERED = "Import sources within a block must be alphabetized.";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        const orderedImportsWalker = new OrderedImportsWalker(sourceFile, this.getOptions());
+        return this.applyWithWalker(orderedImportsWalker);
+    }
+}
+
+// Convert aBcD --> AbCd
+function flipCase(x: string): string {
+    return x.split("").map(char => {
+        if (char >= "a" && char <= "z") {
+            return char.toUpperCase();
+        } else if (char >= "A" && char <= "Z") {
+            return char.toLowerCase();
+        }
+        return char;
+    }).join("");
+}
+
+// Are the nodes sorted according to the text they contain?
+// If not, return the pair of nodes which are out of order.
+function findUnsortedPair(xs: ts.Node[]): [ts.Node, ts.Node] {
+    for (let i = 1; i < xs.length; i++) {
+        if (flipCase(xs[i].getText()) < flipCase(xs[i - 1].getText())) {
+            return [xs[i - 1], xs[i]];
+        }
+    }
+    return null;
+}
+
+// Returns "./utils" given the tree for, e.g. "import * as utils from "./utils";"
+function findFromSource(node: ts.Node): string {
+    const num = node.getChildCount();
+    for (let i = 0; i < num - 1; i++) {
+        const child = node.getChildAt(i);
+        if (child.kind !== ts.SyntaxKind.FromKeyword) {
+            continue;
+        }
+        const next = node.getChildAt(i + 1);
+        if (next.kind !== ts.SyntaxKind.StringLiteral) {
```

Comment:
I appreciate this explicit handling of errors, but it turns out its normally better to silently continue if something malformed is found. Reason being that when people use TSLint in their editor, their code gets linted as they type and the resulting AST isn't always completely well-formed. We don't really want TSLint to start throwing errors here though, it's better for it to just move on.